.github/workflows/update-cidrs.yaml

```diff
@@ -11,42 +11,35 @@ jobs:
     name: Update automatically generated files
     runs-on: ubuntu-latest
     steps:
-      - uses: actions/checkout@v3
-      - uses: actions/setup-node@v3
+      - uses: actions/create-github-app-token@v1
+        id: app-token
         with:
-          node-version: ^18.x
-          cache: 'npm'
+          app-id: ${{ vars.DEVTOOLS_BOT_APP_ID }}
+          private-key: ${{ secrets.DEVTOOLS_BOT_PRIVATE_KEY }}
 
-      - name: Install npm@8
-        run: |
-          npm install -g npm@8
+      - uses: actions/checkout@v4
+        with:
+          # don't checkout a detatched HEAD
+          ref: ${{ github.head_ref }}
+          token: ${{ steps.app-token.outputs.token }}
+
+      - uses: actions/setup-node@v4
+        with:
+          node-version: ^20.x
```

Comment:
The action has been failing for a while because node 18 is not a supported engine.